config/common/errors.go

```diff
@@ -54,7 +54,8 @@ var (
 	ErrTooFewMirrorDevices     = errors.New("mirroring requires at least two devices")
 
 	// partition
-	ErrWrongPartitionNumber = errors.New("incorrect partition number; a new partition will be created using reserved label")
+	ErrInvalidPartitionConfig = errors.New("unexpected partition number; zero is for root disk or when wipe_tables is set to true. To declare a root disk specify the config.disk.device as /dev/disk/by-id/coreos-boot-disk")
```

Comment:
The first sentence seems to imply that, on the root disk or with `wipe_tables` true, the partition number _should_ be zero.  That's not correct.  Users can set the partition number to zero if they want, but not on non-root disks with `wipe_tables` false.  Could you make another try at this, focusing on specifically why the configuration isn't valid?

Also, instead of `config.disk.device`, `device` is probably clear enough, since the error will be attached to the adjacent partition number field.

Reply:
have it changed to "invalid partition configuration; Partition number should not be zero, except the root disk or when wipe_table is set to true. To declare a root disk specify the device as /dev/disk/by-id/coreos-boot-disk"

wdyt?

Reply:
That fixes the "should be zero" problem, but still doesn't explain _why_ the configuration isn't okay.  Please take another look at #377 and try to summarize.  I can help edit the result.